plugin/tools.py

```diff
@@ -452,3 +482,16 @@ def get_clang_version_str(clang_binary):
         else:
             raise RuntimeError(
                 " Couldn't find clang version in clang version output.")
+
+    @staticmethod
+    def get_unique_str(init_string):
+        """ Generate md5 unique sting hash given init_string """
+        import hashlib
+        return hashlib.md5(init_string.encode('utf-8')).hexdigest()
+
+    @staticmethod
+    def get_position_hash(view, position_in_file):
+        """ Generate md5 unique sting hash from position in the file """
+        unique_id = Tools.get_unique_str(
+            view.file_name() + str(position_in_file))
```

Comment:
Is hashing really necessary? Maybe just use `'%s:%s' % (view.id(), position_in_file)`? That is both more unique than using file path and also short enough to not need to hash.